core/server/services/themes/validate.js

```diff
@@ -31,7 +33,11 @@ checkTheme = function checkTheme(theme, isZip) {
 
             return Promise.reject(new common.errors.ThemeValidationError({
                 message: common.i18n.t('errors.api.themes.invalidTheme'),
-                errorDetails: checkedTheme.results.error,
+                errorDetails: Object.assign(
+                    _.pick(checkedTheme, ['checkedVersion', 'name', 'path', 'version']), {
+                        errors: checkedTheme.results.error
+                    }
+                ),
                 context: checkedTheme
```

Comment:
Unfortunately, this whole object has to stay here for now, and [the flow](https://github.com/TryGhost/Ghost/blob/9810834/core/server/services/themes/index.js#L56-L57) that depends on `context` being set to `checkedTheme` needs more investigation :disappointed: